http.go

```diff
@@ -436,9 +441,90 @@ func (h *RasterHttpServer) handleDocumentInfo(w http.ResponseWriter, docParams *
 	w.Write(data)
 }
 
+func writeImage(w http.ResponseWriter, image image.Image, imgParams *RasterImageParams) error {
+	if imgParams.ImageType == "image/jpeg" {
+		return jpeg.Encode(w, image, &jpeg.Options{Quality: imgParams.ImageQuality})
+	}
+
+	return png.Encode(w, image)
+}
+
+// supportsGzipEncoding makes sure that we don't have any false positives
+// Inspired from https://groups.google.com/d/msg/golang-nuts/NVnqAzKbrKQ/6S9wR_zdg4EJ
+func supportsGzipEncoding(req *http.Request) bool {
+	for _, v := range strings.Split(strings.ToLower(req.Header.Get("Accept-Encoding")), ",") {
+		if strings.ToLower(strings.TrimSpace(v)) == "gzip" {
+			return true
+		}
+	}
+	return false
+}
+
+// acquireGzipWriter tries to return a cached gzip.Writer. It will create a new one
+// if none are available.
+func acquireGzipWriter(w http.ResponseWriter) (*gzip.Writer, error) {
+	cachedObject := gzipWriterPool.Get()
+	if cachedObject == nil {
+		return gzip.NewWriter(w), nil
+	}
+
+	gzw := cachedObject.(*gzip.Writer)
+	gzw.Reset(w)
+
+	return gzw, nil
+}
+
+// releaseGzipWriter returns a closed gzip.Writer to the gzip writer pool for reuse
+func releaseGzipWriter(gzw *gzip.Writer) error {
+	err := gzw.Close()
+
+	// I think it might be a bad idea to cache writers which errored when
+	// we tried to close them.
+	if err != nil {
+		return err
+	}
+
+	gzipWriterPool.Put(gzw)
+
+	return nil
+}
+
+func writeSVG(w http.ResponseWriter, r *http.Request, svg []byte) error {
+	var err error
+	if supportsGzipEncoding(r) {
+		var gzw *gzip.Writer
+		gzw, err = acquireGzipWriter(w)
+		if err != nil {
+			return fmt.Errorf("failed to acquire gzip writer: %s", err)
+		}
+
+		w.Header().Add("Content-Encoding", "gzip")
+		// Allow intermediary services to cache different encodings for the same request
+		w.Header().Add("Vary", "Accept-Encoding")
+
+		// Closing the writer can sometimes return an error. We want to return this
+		// error if we don't already have an error from another place in this function.
+		defer func() {
+			errClose := releaseGzipWriter(gzw)
+			if err == nil {
```

Comment:
Not sure, but it looks like you want to test `errClose` returned by `releaseGzipWriter()` instead `err` here, no?

```go
if errClose != nil {
```

Reply:
Nice catch, thanks @sbz! That whole error checking logic was busted. I added a few tests for it.